### DISCUSSION TITLE
Field Resolver Not Excluding Relations

### DISCUSSION THREAD

**Discussion by rtnolan** [MichalLytek/typegraphql-prisma#112] [Q&A] [answered]
OS: Mac OS (Catalina)
type-graphql version: 1.0.0-rc.1
typegraphql-prisma: 0.8.2
prisma/cli version: 2.8.1 (I know this is quite an older version, haven't got around to adopting the newer Prisma version yet).
\---------------------------------------------------

Hi there, I have the following Field Resolver for a relation on an output type:
```javascript
@FieldResolver((_type) => [Position], { nullable: true, description: undefined })
    async Positions(@Root() portfolio: PortfolioOutput, @Args() args: PortfolioPositionsArgs, @Ctx() ctx: Context): Promise<Position[] | null> {
        return ctx.prisma.portfolio
            .findOne({
                where: {
                    id: portfolio.id
                }
            })
            .Positions({
                include: {
                    Transactions: {
                        where: {
                            action: "Buy"
                        }
                    }
                }
            });
    }
```
When I execute my query I'd have expected to have all Transactions that don't have an _action_ of "Buy" to be removed but that isn't the case (see screenshot). Am I missing something? Any help would be appreciated.

![image](https://user-images.githubusercontent.com/11179798/117788380-42e63900-b23f-11eb-98e3-2c9f8d8e306a.png)

**Comment by MichalLytek** [comment, marked as answer]
You can't filter `Position.Transactions` on `Portfolio.Positions` level.

`Position.Transactions` is a separate field resolver that doesn't use the data returned from the other resolver, just the `Position.id` primary key.

You can only filter `Position` that doesn't have any or all "Buy" Transactions.

**Reply by rtnolan** [reply]
Okay cool! I'm assuming what I'll need to do is also have a custom field resolver for _Positions_ then where I will be able to filter them out. Thanks for your help.
